File: kmp-workshop/composeApp/src/androidMain/kotlin/com/chmelik/kmp_workshop/BookDetailView.kt
package com.chmelik.kmp_workshop

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp

@Composable
fun BookDetailView(
    book: Book,
    viewModel: BookViewModel
) {
    Column(Modifier.padding(16.dp)) {
        Text(text = book.name, fontWeight = FontWeight.Bold)
        Text(text = "Pages: ${book.pageCount}")
        Text(text = "Rating: ${book.rating}")
        Button(
            onClick = {
                viewModel.likeUnlikeBook(book.id)
            }
        ) {
            Text(if (viewModel.isBookLiked(book.id)) "Unlike" else "Like")
        }
    }
}

File: kmp-workshop/iosApp/iosApp/BookDetailView.swift
//
// Created by Dominik Chmelík on 15.10.2025.
//

import SwiftUI

struct BookDetailView: View {
    @Binding var viewModel: BookViewModel
    let book: BookItem

    var body: some View {
        VStack(spacing: 10) {
            Text(book.name)
                .fontWeight(.heavy)
                .padding(.bottom, 10)
            Text("\(book.pageCount) pages")
            Text("Rating \(book.rating)/5")
            Button {
                viewModel.likeUnlikeBook(id: book.id)
            } label: {
                let isLiked = viewModel.isBookLiked(id: book.id)
                Image(systemName: isLiked ? "heart.fill" : "heart")
                    .tint(isLiked ? .red : .gray)
            }
        }
    }
}
